pkg/admissioncontroller/webhook/auth/seed/graph/graph.go

```diff
@@ -87,6 +87,8 @@ func (g *graph) Setup(ctx context.Context, c cache.Cache) error {
 }
 
 func (g *graph) HasVertex(vertexType VertexType, vertexNamespace, vertexName string) bool {
+	g.lock.RLock()
+	defer g.lock.RUnlock()
```

Comment:
Why not add this to `getVertex`?

Reply:
The lock is a `sync.RWMutex` and not reentrant. In all calling methods, `getVertex` is only used after the lock has been applied. The only exception is the call in `HasVertex`.